xla/stream_executor/command_buffer.h

```diff
@@ -172,141 +105,126 @@ class CommandBuffer {
   // Command buffer API
   //===--------------------------------------------------------------------===//
 
-  // Adds an execution barrier to a given execution scope: all commands added
-  // before a barrier in a the execution scope will complete before any of the
-  // commands added after a barrier in the same execution scope.
-  virtual absl::Status Barrier(ExecutionScopeId execution_scope_id) = 0;
+  // Adds an execution barrier that depends on the commands in dep_idxes.
+  virtual absl::Status EmptyOp(Index cmd_idx, const CmdIndexSet& dep_idxes) = 0;
```

Comment:
dep_idxes -> deps (or dependencies) everywhere, the fact that it's indices is clear from the type name